doc/mediasdk-man.md

```diff
@@ -1161,6 +1165,125 @@ Assume the pipeline is **A->B->C**. The application synchronizes on sync point *
 
 ## <a id='hardware_acceleration'>Working with hardware acceleration</a>
 
+### <a id='Working_with_multiple_intel_media_devices'>Working with multiple Intel media devices</a>
+
+If your system has multiple Intel Gen Graphics adapters you may need hints on which adapter suits better to process some particular workload. The SDK provides API to help with adapter selection. See Example 9 that showcases workload processing initialization on discrete adapter.
+
+###### Example 9: Choose discrete Intel Gen Graphics adapter
+
+```C
+mfxU32 num_adapters_available;
+
+// Query number of Intel Gen Graphics adapters available on system
+mfxStatus sts = MFXQueryAdaptersNumber(&num_adapters_available);
+MSDK_CHECK_STATUS(sts, "MFXQueryAdaptersNumber failed");
+
+// Allocate memory for response
+std::vector<mfxAdapterInfo> displays_data(num_adapters_available);
+mfxAdaptersInfo adapters = { displays_data.data(), mfxU32(displays_data.size()), 0u };
+
+// Query information about all adapters (mind that first parameter is NULL)
+sts = MFXQueryAdapters(nullptr, &adapters);
+MSDK_CHECK_STATUS(sts, "MFXQueryAdapters failed");
+
+// Find dGfx adapter in list of adapters
+auto idx_d = std::find_if(adapters.Adapters, adapters.Adapters + adapters.NumActual,
+    [](const mfxAdapterInfo info)
+{
+    return info.Platform.MediaAdapterType == mfxMediaAdapterType::MFX_MEDIA_DISCRETE;
+});
+
+// No dGfx in list
+if (idx_d == adapters.Adapters + adapters.NumActual)
+{
+    printf("Warning: No dGfx detected on machine\n");
+    return -1;
+}
+
+mfxU32 idx = static_cast<mfxU32>(std::distance(adapters.Adapters, idx));
+
+// Choose correct implementation for discrete adapter
+switch (adapters.Adapters[idx].Number)
```

Comment:
hm. And why API does not return impl directly:)?

Reply:
Because impl type supports up to 4 adapters, in future we will use ext buffer to support more adapters